Rubberduck.Parsing/ComReflection/DeclarationsFromComProjectLoader.cs

```diff
@@ -0,0 +1,307 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Parsing.ComReflection
+{
+    public class DeclarationsFromComProjectLoader : IDeclarationsFromComProjectLoader
+    {
+        public IReadOnlyCollection<Declaration> LoadDeclarations(ComProject type, string projectId = null)
```

Comment:
From the API and the implementation I see no particular reason to make this an instance method. The only benefit we get is the injection possibility... As a component this would be testable anyways... So this is a class to allow injecting our own Declarations into the usages for testing?

Reply:
This class is a strategy and not static to allow mocking it in tests for its consumers.